File: homeworks/hw1/gcd_full.py
#! /usr/bin/env python
import sys


def gcdI(i, j):
    while i != j:
        if i > j:
            i = i - j
        else:
            j = j - i
    return i


def gcdF(i, j):
    if not i % j: return j 
    return gcdF(j, i % j)


if len(sys.argv) != 3:
    print(sys.argv[0] + " usage: [I] [J]")
    exit()

gcd = gcdI(int(sys.argv[1]), int(sys.argv[2]))
print(gcd)

File: homeworks/hw1/gcd_full.rb
def gcdI(i, j)
	while i != j do
		if i > j
			i = i - j
		else
			j = j - i
		end
	end
	return i
end


def gcdF(i, j)
	  return j if !(i % j)
		return gcdF(j, i % j)
end


if ARGV.length != 2 
	  puts "gcd_full.rb usage: [I] [J]" 
		  exit
end

puts gcdI(ARGV[0].to_i, ARGV[1].to_i)
